src/vs/code/electron-main/window.ts

```diff
@@ -426,6 +430,11 @@ export class VSCodeWindow {
 		delete configuration.filesToCreate;
 		delete configuration.filesToDiff;
 
+		// Update untitled files to restore so they come through in the reloaded window
```

Comment:
I filed https://github.com/Microsoft/vscode/issues/15498 to continue the discussion